<?xml version="1.0" encoding="utf-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from panda_arm_hand.srdf.xacro      | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<!--This does not replace URDF, and is not an extension of URDF.
    This is a format for representing semantic information about the robot structure.
    A URDF file must exist for this robot as well, where the joints and the links that are referenced are defined
-->
<robot name="open_manipulator_x">
  <!--GROUPS: Representation of a set of joints and links. This can be useful for specifying DOF to plan for, defining arms, end effectors, etc-->
  <!--LINKS: When a link is specified, the parent joint of that link (if it exists) is automatically included-->
  <!--JOINTS: When a joint is specified, the child link of that joint (which will always exist) is automatically included-->
  <!--CHAINS: When a chain is specified, all the links along the chain (including endpoints) are included in the group. Additionally, all the joints that are parents to included links are also included. This means that joints along the chain and the parent joint of the base link are included in the group-->
  <!--SUBGROUPS: Groups can also be formed by referencing to already defined group names-->
  <group name="open_manipulator_x_arm">
    <chain base_link="link1" tip_link="link5"/>
  </group>
  <!--GROUP STATES: Purpose: Define a named state for a particular group, in terms of joint values. This is useful to define states like 'folded arms'-->
  <group_state group="open_manipulator_x_arm" name="ready">
    <joint name="joint1" value="0"/>
    <joint name="joint2" value="0"/>
    <joint name="joint3" value="0"/>
    <joint name="joint4" value="0"/>
  </group_state>

  <!--END EFFECTOR: Purpose: Represent information about an end effector.-->
  <!--VIRTUAL JOINT: Purpose: this element defines a virtual joint between a robot link and an external frame of reference (considered fixed with respect to the robot)-->
  <!-- <virtual_joint child_link="link1" name="virtual_joint" parent_frame="world" type="floating"/> -->

  <disable_collisions link1="link1" link2="link2" reason="Adjacent"/>
  <disable_collisions link1="link2" link2="link3" reason="Adjacent"/>
  <disable_collisions link1="link3" link2="link4" reason="Adjacent"/>
  <disable_collisions link1="link4" link2="link5" reason="Adjacent"/>
  <disable_collisions link1="link5" link2="gripper_link_sub" reason="Adjacent"/>
  <disable_collisions link1="link5" link2="gripper_link" reason="Adjacent"/>

  <!--DISABLE COLLISIONS: By default it is assumed that any link of the robot could potentially come into collision with any other link in the robot. This tag disables collision checking between a specified pair of links. -->
  <!-- <disable_collisions link1="panda_link0" link2="panda_link1" reason="Adjacent"/>
  <disable_collisions link1="panda_link0" link2="panda_link2" reason="Never"/>
  <disable_collisions link1="panda_link0" link2="panda_link3" reason="Never"/>
  <disable_collisions link1="panda_link0" link2="panda_link4" reason="Never"/>
  <disable_collisions link1="panda_link1" link2="panda_link2" reason="Adjacent"/>
  <disable_collisions link1="panda_link1" link2="panda_link3" reason="Never"/>
  <disable_collisions link1="panda_link1" link2="panda_link4" reason="Never"/>
  <disable_collisions link1="panda_link2" link2="panda_link3" reason="Adjacent"/>
  <disable_collisions link1="panda_link2" link2="panda_link4" reason="Never"/>
  <disable_collisions link1="panda_link2" link2="panda_link6" reason="Never"/>
  <disable_collisions link1="panda_link3" link2="panda_link4" reason="Adjacent"/>
  <disable_collisions link1="panda_link3" link2="panda_link5" reason="Never"/>
  <disable_collisions link1="panda_link3" link2="panda_link6" reason="Never"/>
  <disable_collisions link1="panda_link3" link2="panda_link7" reason="Never"/>
  <disable_collisions link1="panda_link4" link2="panda_link5" reason="Adjacent"/>
  <disable_collisions link1="panda_link4" link2="panda_link6" reason="Never"/>
  <disable_collisions link1="panda_link4" link2="panda_link7" reason="Never"/>
  <disable_collisions link1="panda_link5" link2="panda_link6" reason="Adjacent"/>
  <disable_collisions link1="panda_link6" link2="panda_link7" reason="Adjacent"/> -->
  <!--GROUPS: Representation of a set of joints and links. This can be useful for specifying DOF to plan for, defining arms, end effectors, etc-->
  <!--LINKS: When a link is specified, the parent joint of that link (if it exists) is automatically included-->
  <!--JOINTS: When a joint is specified, the child link of that joint (which will always exist) is automatically included-->
  <!--CHAINS: When a chain is specified, all the links along the chain (including endpoints) are included in the group. Additionally, all the joints that are parents to included links are also included. This means that joints along the chain and the parent joint of the base link are included in the group-->
  <!--SUBGROUPS: Groups can also be formed by referencing to already defined group names-->
  <group name="hand">
    <link name="gripper_link"/>
    <link name="gripper_link_sub"/>
    <joint name="gripper"/>
    <passive_joint name="gripper_sub"/>
  </group>

  <disable_collisions link1="gripper_link" link2="gripper_link_sub" reason="Adjacent"/>
  <!--GROUP STATES: Purpose: Define a named state for a particular group, in terms of joint values. This is useful to define states like 'folded arms'-->
  <!--END EFFECTOR: Purpose: Represent information about an end effector.-->
  <!--DISABLE COLLISIONS: By default it is assumed that any link of the robot could potentially come into collision with any other link in the robot. This tag disables collision checking between a specified pair of links. -->
  <!-- <disable_collisions link1="panda_hand" link2="panda_leftfinger" reason="Adjacent"/>
  <disable_collisions link1="panda_hand" link2="panda_rightfinger" reason="Adjacent"/>
  <disable_collisions link1="panda_leftfinger" link2="panda_rightfinger" reason="Default"/> -->
  <!--GROUPS: Representation of a set of joints and links. This can be useful for specifying DOF to plan for, defining arms, end effectors, etc-->
  <!--LINKS: When a link is specified, the parent joint of that link (if it exists) is automatically included-->
  <!--JOINTS: When a joint is specified, the child link of that joint (which will always exist) is automatically included-->
  <!--CHAINS: When a chain is specified, all the links along the chain (including endpoints) are included in the group. Additionally, all the joints that are parents to included links are also included. This means that joints along the chain and the parent joint of the base link are included in the group-->
  <!--SUBGROUPS: Groups can also be formed by referencing to already defined group names-->
  <!-- <group name="panda_arm_hand">
    <group name="panda_arm"/>
    <group name="hand"/>
  </group>
  <group_state group="hand" name="open">
    <joint name="panda_finger_joint1" value="0.035"/>
    <joint name="panda_finger_joint2" value="0.035"/>
  </group_state>
  <group_state group="hand" name="close">
    <joint name="panda_finger_joint1" value="0"/>
    <joint name="panda_finger_joint2" value="0"/>
  </group_state> -->
  <!--END EFFECTOR: Purpose: Represent information about an end effector.-->
  <end_effector group="hand" name="hand" parent_group="open_manipulator_x_arm" parent_link="link5"/>
  <!--DISABLE COLLISIONS: By default it is assumed that any link of the robot could potentially come into collision with any other link in the robot. This tag disables collision checking between a specified pair of links. -->
  <!-- <disable_collisions link1="panda_hand" link2="panda_link3" reason="Never"/>
  <disable_collisions link1="panda_hand" link2="panda_link4" reason="Never"/>
  <disable_collisions link1="panda_hand" link2="panda_link6" reason="Never"/>
  <disable_collisions link1="panda_hand" link2="panda_link7" reason="Adjacent"/>
  <disable_collisions link1="panda_leftfinger" link2="panda_link3" reason="Never"/>
  <disable_collisions link1="panda_leftfinger" link2="panda_link4" reason="Never"/>
  <disable_collisions link1="panda_leftfinger" link2="panda_link6" reason="Never"/>
  <disable_collisions link1="panda_leftfinger" link2="panda_link7" reason="Never"/>
  <disable_collisions link1="panda_link3" link2="panda_rightfinger" reason="Never"/>
  <disable_collisions link1="panda_link4" link2="panda_rightfinger" reason="Never"/>
  <disable_collisions link1="panda_link6" link2="panda_rightfinger" reason="Never"/>
  <disable_collisions link1="panda_link7" link2="panda_rightfinger" reason="Never"/> -->
</robot>

